sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -935,40 +936,132 @@ public void testUnionWithSubSelectRewritten() throws Exception {
                                                    "limit 10 offset 20");
         assertThat(analysis.relation(), instanceOf(TwoRelationsUnion.class));
         TwoRelationsUnion tableUnion = (TwoRelationsUnion) analysis.relation();
-        assertThat(tableUnion.first(), instanceOf(QueriedDocTable.class));
-        assertThat(tableUnion.second(), instanceOf(QueriedDocTable.class));
+        assertThat(tableUnion.first(), instanceOf(QueriedSelectRelation.class));
+        assertThat(tableUnion.second(), instanceOf(QueriedSelectRelation.class));
         assertThat(tableUnion.querySpec(), isSQL("SELECT doc.users.id " +
                                                  "ORDER BY INPUT(0) " +
                                                  "LIMIT 10 OFFSET 20"));
-        assertThat(tableUnion.first().querySpec(), isSQL("SELECT doc.users.id " +
-                                                         "ORDER BY doc.users.id " +
-                                                         "LIMIT least(5, add(10, 20))"));
+        QueriedSelectRelation firstRelation = ((QueriedSelectRelation) tableUnion.first());
+        assertThat(firstRelation.querySpec(), isSQL("SELECT doc.users.id " +
+                                                    "ORDER BY doc.users.id " +
+                                                    "LIMIT add(10, 20)"));
+        assertThat(firstRelation.subRelation().querySpec(), isSQL("SELECT doc.users.id, doc.users.name " +
+                                                                  "ORDER BY doc.users.name " +
+                                                                  "LIMIT 5"));
         assertThat(tableUnion.second().querySpec(), isSQL("SELECT doc.users_multi_pk.id " +
                                                           "ORDER BY doc.users_multi_pk.id " +
                                                           "LIMIT add(10, 20)"));
     }
 
     @Test
-    public void testUnionAsSubSelect() throws Exception {
-        expectedException.expect(UnsupportedOperationException.class);
-        expectedException.expectMessage("UNION as a sub query is not supported");
-        analyze("select id, name from (" +
-                    "select id, name, text from users " +
-                    "union all " +
-                    "select id, name, name from users_multi_pk " +
-                    "order by id " +
-                    "limit 10 offset 20) a " +
-                "order by 2 limit 5") ;
+    public void testUnionAsSubSelectRewritten() throws Exception {
+        SelectAnalyzedStatement analysis = analyze("select id, text from (" +
+                                                     "select id, name, text from users where id = 1 " +
+                                                     "union all " +
+                                                     "select id, name, name as text from users_multi_pk where id = 2 " +
+                                                     "order by id) a " +
```

Comment:
maybe move closing bracket to extra line for readability (so its more obvious where the subselect ends). consider doing that at all tests here